modular_zubbers/code/modules/surgery/surgery_step.dm

```diff
@@ -14,3 +14,31 @@
 			return TRUE
 
 	return FALSE
+
+/datum/surgery_step/success(mob/user, mob/living/target, target_zone, obj/item/tool, datum/surgery/surgery, default_display_results = TRUE)
+	SEND_SIGNAL(user, COMSIG_MOB_SURGERY_STEP_SUCCESS, src, target, target_zone, tool, surgery, default_display_results)
+	if(default_display_results)
+		display_results(
+			user,
+			target,
+			span_notice("You succeed."),
+			span_notice("[user] succeeds!"),
+			span_notice("[user] finishes."),
+		)
+	if(ishuman(user))
+		var/mob/living/carbon/human/surgeon = user
+		if (ishuman(target))
+			var/mob/living/carbon/human/human_target = target
+			var/obj/item/bodypart/target_bodypart = target.get_bodypart(target_zone)
+			var/obj/item/organ/eyes/target_eyes = target.get_organ_slot(ORGAN_SLOT_EYES)
+			if(target_bodypart)
+				if(target_bodypart.bodytype != BODYTYPE_ROBOTIC && !HAS_TRAIT(human_target, TRAIT_NOBLOOD))
+					surgeon.add_blood_DNA_to_items(target.get_blood_dna_list(), ITEM_SLOT_GLOVES)
+			else if(target_eyes) // snowflake case for eyes
+				if(target_eyes.organ_flags != ORGAN_ROBOTIC && !HAS_TRAIT(human_target, TRAIT_NOBLOOD))
+					surgeon.add_blood_DNA_to_items(target.get_blood_dna_list(), ITEM_SLOT_GLOVES)
+		else
+			surgeon.add_blood_DNA_to_items(target.get_blood_dna_list(), ITEM_SLOT_GLOVES)
+	else
```

Comment:
Sorry, I know I just told you to modularize, but for instances like this where you're adding something to a proc its usually better to add that part non-modularly, just do `// BUBBER EDIT ADDITION BEGIN` and `// BUBBER EDIT ADDITION END` and put it in between in the original file. Reason being, for something like this, if we paste the whole proc into a modular file we fully override that, and if upstream ever changes how this proc works we have no real way of knowing, it'll just keep running ours. Everything else looks good though!